library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -207,7 +222,10 @@ void resume() {
     }
 
     void delete() {
-        downloadsBatchPersistence.deleteAsync(downloadBatchStatus.getDownloadBatchId());
+        if (downloadBatchStatus.status() == PAUSED) {
+            downloadsBatchPersistence.deleteAsync(downloadBatchStatus.getDownloadBatchId());
```

Comment:
Just to check, do we still need this to be async?

Reply:
yes in this case, because this `delete()` is an immediate request (main thread)